### PR TITLE
Add an option for service to send parameteres

### PR DESCRIPTION
**Pull request by musayann**
Added an option in service model to allow get parameters to be sent
overrode the index function in recommendation to allow it to send iteration id

### CODE REVIEW

File: app/javascript/models/Recommendation.js [hunk 1]
```diff
@@ -1,5 +1,12 @@
 import BaseModel from "./BaseModel";
+import Service from "./Service";
 
 export default class Recommendation extends BaseModel {
      static basePath = "recommendations";
+
+     static index(id) {
+       return Service.get(this.url(), {
+         'iteration_id': id
```
Comment (fsmanuel) [inline app/javascript/models/Recommendation.js]: Why do we need it?

File: app/javascript/models/Service.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
 export default class Service {
   static get(url, params = {}) {
-    return this.request('get', url, params);
+    return this.request('get', url, {}, params);
```
Comment (fsmanuel) [inline app/javascript/models/Service.js]: I think there should not be data for a get request.

File: app/javascript/models/Service.js [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
   }
 
   static post(url, params = {}) {
```
Comment (fsmanuel) [inline app/javascript/models/Service.js]: we should rename `params` to `data` here.

File: app/javascript/models/Recommendation.js (hunk 1 above)
Comment (musayann) [reply app/javascript/models/Recommendation.js]: To get recommendation, the iteration_id is required and the route is pointing to recommendations#index

File: app/javascript/models/Service.js (hunk 3 above)
Comment (musayann) [reply app/javascript/models/Service.js]: The problem is with axios, you can't send data with a get request, but you can send params with get request

**Comment by musayann** [comment]
> Can you explain why we need that changes?

We were unable to get recommendations since the `recommendations#index` requires the parameter iteration_id and the route is get, and also the problem is with Axios, you can't send data with a get request, but you can send params with get request

File: app/javascript/models/Service.js (hunk 3 above)
Comment (fsmanuel) [reply app/javascript/models/Service.js]: That is what I'm saying. It's the `post` method and it calls the `request` method where the arguments name is `data`.

File: app/javascript/models/Service.js (hunk 2 above)
Comment (fsmanuel) [reply app/javascript/models/Service.js]: Instead of `{}` we should do `null`

File: app/javascript/models/Recommendation.js (hunk 1 above)
Comment (fsmanuel) [reply app/javascript/models/Recommendation.js]: Ok, now I see the problem. I was just looking into the recommendations controller and routes and couldn't see any `iteration_id`. So we need it to fetch recommendations for a specific iteration defined by `current_iteration`, right?

File: app/javascript/models/Recommendation.js [hunk 4]
```diff
@@ -1,5 +1,12 @@
 import BaseModel from "./BaseModel";
+import Service from "./Service";
 
 export default class Recommendation extends BaseModel {
      static basePath = "recommendations";
+
+     static index(id) {
```
Comment (fsmanuel) [inline app/javascript/models/Recommendation.js]: I think it's cleaner to supply the full options here or rename `id` to `iterationId`.

```js
static index(options = null) {
  return Service.get(this.url(), options);
}
```

File: app/javascript/models/Service.js (hunk 2 above)
Comment (fsmanuel) [reply app/javascript/models/Service.js]: Sorry for the last comment that doesn't work. You can resolve this conversation.

**Pull request merged**